awx/ui_next/src/components/Sort/Sort.jsx

```diff
@@ -17,59 +19,47 @@ import {
   SortNumericUpIcon,
 } from '@patternfly/react-icons';
 
+import { parseQueryString } from '@util/qs';
+import { SortColumns, QSConfig } from '@types';
 import styled from 'styled-components';
 
-const Dropdown = styled(PFDropdown)`
-  &&& {
-    > button {
-      min-height: 30px;
-      min-width: 70px;
-      height: 30px;
-      padding: 0 10px;
-      margin: 0px;
-
-      > span {
-        /* text element within dropdown */
-        width: auto;
-      }
-
-      > svg {
-        /* caret icon */
-        margin: 0px;
-        padding-top: 3px;
-        padding-left: 3px;
-      }
-    }
-  }
-`;
-
-const IconWrapper = styled.span`
-  > svg {
-    font-size: 18px;
-  }
-`;
-
-const SortButton = styled(Button)`
-  padding: 5px 8px;
-  margin-top: 3px;
-
-  &:hover {
-    background-color: #0166cc;
-    color: white;
-  }
-`;
-
-const SortBy = styled.span`
-  margin-right: 15px;
-  font-size: var(--pf-global--FontSize--md);
+const NoOptionDropdown = styled.div`
+  align-self: stretch;
+  border: 1px solid var(--pf-global--BorderColor--300);
+  padding: 5px 15px;
+  white-space: nowrap;
+  border-bottom-color: var(--pf-global--BorderColor--200);
 `;
 
 class Sort extends React.Component {
   constructor(props) {
     super(props);
 
+    let sortKey;
+    let sortOrder;
+    let isNumeric;
+
+    const { qsConfig, location } = this.props;
+    const queryParams = parseQueryString(qsConfig, location.search);
+    if (queryParams.order_by && queryParams.order_by.startsWith('-')) {
+      sortKey = queryParams.order_by.substr(1);
+      sortOrder = 'descending';
+    } else if (queryParams.order_by) {
+      sortKey = queryParams.order_by;
+      sortOrder = 'ascending';
+    }
+
+    if (qsConfig.integerFields.filter(field => field === sortKey).length) {
```

Comment:
We're doing this logic in a few places. It's probably worth adding a helper function to qs (something like `isFieldNumeric(qsConfig, fieldName)`). It's a little cleaner to use `Array.find()`instead of `.filter()`, too.

It might be worth doing the same with the `sortOrder` check above, too.

Reply:
I can update to find, but I don't think I agree that moving this up a level is an improvement.

`qsConfig.integerFields.find(field => field === sortKey)` vs `qsConfig.isFieldNumeric(qsConfig, sortKey)`

I don't find one cleaner than the other, and I don't think making the qs API more complex for this case is worth it.